crates/hir-def/src/find_path.rs

```diff
@@ -551,7 +551,18 @@ fn find_local_import_locations(
         if let Some((name, vis)) = data.scope.name_of(item) {
             if vis.is_visible_from(db, from) {
                 let is_private = match vis {
-                    Visibility::Module(private_to) => private_to.local_id == module.local_id,
+                    Visibility::Module(private_mod, private_meta) => {
+                        if private_mod == def_map.module_id(DefMap::ROOT)
```

Comment:
Shouldn't it be enough to just check this? I don't see why we need the `is_pub_crate`. If the `private_mod` here is a crate root module then it already has to be `pub(crate)` or something that has t he same semantic meaning.

Reply:
Ah I see, yes that would definitely be unexpected. That deserves an extra test to check that we aren't doing that for that matter! (assuming there is no such test yet).

Reply:
Although, what is rustc's behavior in regards to that. Does it also only acknowledge `pub(crate)` re-exports for this?

Reply:
rustc _does_ suggest `use crate::Foo;`, but I'm not quite sold on it 😅

Reply:
Ye me neither, so let's stick to checking whether the visibility is explicitly provided.